torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -198,21 +208,49 @@ Tensor pow_backward_self(Tensor grad, const Tensor & self, const Tensor & expone
 // d(a^b)/db = 0 for a > 0 and b -> +0.
 // Currently, tensorflow agrees with us.
 Tensor pow_backward_exponent(Tensor grad, const Tensor& self, const Tensor& exponent, Tensor result) {
-  return grad * at::where(at::logical_and(self == 0, exponent >= 0),
+  auto cond = [](auto self, auto exp) {
```

Comment:
Why did you put this in a lambda?
Same in the function below.

Reply:
The condition input to `at::where` is different for complex and non-complex inputs. I thought it's better to put it in a lambda function as opposed to global scope since it's small and only used once.

Reply:
But this lambda is never re-used and just called once one line below. So why not just remove it just use the code inside directly and pass the computed Tensor directly to the next line?

Reply:
But this lambda is only ever called once on the next line. So why not just leave the content without the lambda and use the computed Tensor on the next line?

Reply:
makes sense will do